Documentation/upgrade.md

```diff
@@ -131,350 +181,165 @@ Any pod that is using a Rook volume should also remain healthy:
+Ceph mgr daemon. While we continue to investigate this issue, if you see two mgr pods running,
+delete the old mgr replicaset in order for the new Ceph mgr to start up properly.
+
+This has been observed on Kubernetes version v1.9.8. If you experience the issue, please let us know
+on [GitHub (#2337)](https://github.com/rook/rook/issues/2337) which Kubernetes version you are
+running to better help us resolve the issue.
+
+### 6. Remove old resources
+Once the cluster is done updating, resources removed in v0.9 can be safely removed.
+```sh
+kubectl -n $ROOK_NAMESPACE delete serviceaccount rook-ceph-cluster
+kubectl -n $ROOK_NAMESPACE delete role           rook-ceph-cluster
+kubectl -n $ROOK_NAMESPACE delete rolebinding    rook-ceph-cluster
+```
+
+### 7. Verify the updated cluster
+At this point, your Rook operator should be running version `rook/ceph:v0.9.0`, and the Ceph daemons
+should be running image `ceph/ceph:v12.2.9-20181026`. The Rook operator version and the Ceph version
+are no longer tied together, and we'll cover how to upgrade Ceph later in this document.
+
+Verify the Ceph cluster's health using the [health verification section](#health-verification).
+
+### 8. Update optional components
+In general, ancillary components of Rook can be updated using the
+update-crd-then-create-then-replace pattern from [step 2](#2.-add-new-resources) and
+[step 3](#3.-update-existing-resources). Some specific notes can be found below.
+
+#### File and object storage
+There have been no significant changes to the file or object storage CRDs, so updating these should
+be unnecessary.
+
+#### Ceph dashboard
+There have been some changes to the Ceph dashboard in Mimic which affect Rook. In Ceph Luminous
+(ceph:v12), the dashboard uses HTTP on port 7000 by default. In Ceph Mimic (ceph:v13), the dashboard
+uses HTTPS on port 8443 by default. To upgrade the dashboard external service, the v0.8 service must
+be deleted and the appropriate new service added.
+```sh
+kubectl -n $ROOK_NAMESPACE delete service rook-ceph-mgr-dashboard-external
+kubectl -n $ROOK_NAMESPACE create -f dashboard-external-http.yaml  # for Ceph Luminous (v12)
+```
+
+
+## Ceph Upgrades
+By default, an upgraded Rook cluster will be running Ceph Luminous (v12), the same Ceph version
+released with Rook v0.8. Now that the Rook and Ceph versions are independently controllable, you can
+choose to update Ceph at any time.
+
+### Ceph images
+Official Ceph container images can be found on [Docker Hub](https://hub.docker.com/r/ceph/ceph/tags/).
+These images are tagged in a few ways:
+* The most explicit form of tags are full-ceph-version-and-build tags (e.g., `v13.2.2-20181023`).
+  These tags are recommended for production clusters, as there is no possibility for the cluster to
+  be heterogeneous with respect to the version of Ceph running in containers.
+* Ceph minor version tags (e.g. `v13.2`) may be desired when a cluster is non-production but where
```

Comment:
We certainly don't need to. But I'm not sure what I might be missing regarding the minor version being the same as the major tag. `v13.2` is the same as `v13` only until v13.3.0 is released. And `v13.0, v13.1` are not the same as `v13` currently.

Reply:
@BlaineEXE @travisn It is important to use the most explicit tag.
If an user uses tag `v13` (while Ceph v13.0.0 is out) and adds a new node still with image tag `v13` while Ceph v13.10.10 is out, the cluster could break and make it a hell to debug because of the image tag being the same (the shasum changes, but it is a pain to debug still when there are Ceph related problems between those minor versions).

Reply:
@BlaineEXE for ceph the minor version `.2` will always remain the same. So `v13.3.x` will never be released, and `v13.1.x` indicates a testing pre-release.
@galexrt agreed that we want to be explicit about the specific images to avoid that situation. This is why i want to reduce it to only two comments: one for production and the other for development clusters.